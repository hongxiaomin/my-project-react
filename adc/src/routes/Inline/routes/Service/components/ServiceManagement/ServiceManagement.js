import React from 'react'
import { connect } from 'react-redux'
import classes from './ServiceManagement.scss'
import SMList from '../ServiceManagementList'
import SMQueueList from '../QueueList'
import { getModel, getServiceInfo, getQueueList } from '../../modules/Service'

type Props = {
  getModel: Function,
  getServiceInfo: Function,
  getQueueList: Function,
};

export class ServiceManagement extends React.Component {
  componentDidMount() {
    this.props.getServiceInfo()
    this.props.getQueueList()
    this.props.getModel()
  }

  props: Props

  render() {
    return (
      <div className={classes['ServiceManagement-container']}>
        <SMList
          setDetail={this.setDetail}
        />
        <SMQueueList />
      </div>
    )
  }
}

export default connect(
  state => ({
    data: state.service,
  }), {
    getModel,
    getServiceInfo,
    getQueueList,
  })(ServiceManagement)
