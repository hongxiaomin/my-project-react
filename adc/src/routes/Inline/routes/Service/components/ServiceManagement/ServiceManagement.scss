:global {
  .ReactTable .rt-tbody .rt-expandable {
    text-overflow: clip;
  }
  .ReactTable .rt-expander {
    &:after {
      content: '+';
      border: none;
      color: rgba(0, 0, 0, 0.65);
      font-size: 18px;
      margin-left: -5px;
      width: auto;
      height: auto;
      transition: none;
    }
  }
  .ReactTable .rt-expander.-open {
    &:after {
      content: '-';
      margin-left: -3px;
      margin-top: -3px;
    }
  }
}

.ServiceManagement-container {
  padding: 1% 0;
  display: flex;
  justify-content: center;
}
.top-left{
  display: flex;
  justify-content: space-between;
  height: 50px;
  width: 100%;
  line-height: 50px;
  font-size: 16px;
  padding: 0px 30px;
  background-color: #f7f7f7;
  // border: 1px solid #d7d7d7;
  border-radius: 4px;
  margin-top: 20px;

}
