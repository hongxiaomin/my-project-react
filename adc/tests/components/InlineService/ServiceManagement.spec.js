import React from 'react'
import { ServiceManagement } from 'routes/Inline/routes/Service/components/ServiceManagement/ServiceManagement'

describe('(Component) ServiceManagement', () => {
  const getQueueList = jest.fn()
  const getModel = jest.fn()
  const getServiceInfo = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <ServiceManagement
        getQueueList={getQueueList}
        getModel={getModel}
        getServiceInfo={getServiceInfo}
      />,
    )
    // wrapper.find('Button').last().simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
})
