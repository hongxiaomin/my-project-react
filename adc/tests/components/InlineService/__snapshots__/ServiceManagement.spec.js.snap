// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ServiceManagement render normally 1`] = `
<div
  className="ServiceManagement-container"
>
  <Connect(ServiceManagementList) />
  <Connect(QueueList) />
</div>
`;
